client/Components/CodeEditor/CodeEditor.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react'
+import axios from 'axios'
+import Select from '@material-ui/core/Select'
+import OutlinedInput from '@material-ui/core/OutlinedInput'
+import MenuItem from '@material-ui/core/MenuItem'
+import configEditor from './EditorConfigFunction'
+
+class AceEditor extends Component {
+  constructor() {
+    super()
+    this.state = {
```

Comment:
Just to clarify what the state holds:
code: the code that is typed into the editor
codeResponse: what comes back from the server
codeError: errors
editor: essentially, this is where the ace-editor that is embedded into the dom lives

Reply:
Yep you got it all right. Just to clarify b/c I didn't make it very clear in the code, codeError is specvficially for any errors from the user's code but right now I think it would pick up any errors coming from the backend. So that's something to deal with. On a more general note, while I dealt with most things I think, it's not fully formed b/c I most wanted to get this up for you guys to see.

Yep, you are right about the editor. In the render, there's div with id of ace and that's where the editor attaches itself to. On another general note, I did that instead of using the React-Ace component library b/c triyng to follow Eliot and Prof's advice of using libraries only when truly necessary